components/service/glean/src/main/java/mozilla/components/service/glean/storages/GenericScalarStorageEngine.kt

```diff
@@ -37,14 +37,20 @@ internal typealias ScalarRecordingCombiner<T> = (currentValue: T?, newValue: T)
  * A base class for 'scalar' like metrics. This allows sharing the common
  * store managing and lifetime behaviours.
  */
+@Suppress("TooManyFunctions")
 internal abstract class GenericScalarStorageEngine<ScalarType> : StorageEngine {
     override lateinit var applicationContext: Context
 
     // Let derived class define a logger so that they can provide a proper name,
     // useful when debugging weird behaviours.
     abstract val logger: Logger
 
-    protected val userLifetimeStorage: SharedPreferences by lazy { deserializeUserLifetime() }
+    protected val userLifetimeStorage: SharedPreferences by lazy {
+        deserializeLifetime(Lifetime.User)!!
```

Comment:
The `!!` after this is scary. However, given that the function only returns `null` if `Lifetime.Application`, which is not the case here, we should be ok. However, I'd add a comment above this and line 52.